firmware/library/L3_HAL/apds_9960.hpp

```diff
@@ -1,20 +1,444 @@
+  kPoffsetDownLeft = 0x00,
+  kConfig3 = 0x00, kGPEnTh = 0x28,
+  kGExTh = 0x64, kGconfig1 = 0x40,
+  kGconfig2 = 0x66,
+  kGoffsetUpDown = 0x00,
+  kGoffsetLeftRight = 0x20,
+  kGPulse = 0xC5,
+  kGconfig3 = 0x00, kGconfig4 = 0x02;
+
+  void WriteReg(uint8_t reg, uint8_t data) override
+  {
+    uint8_t data_stream[2] = {reg, data};
+    i2c.Write(kApds9960Address, data_stream,
+        sizeof(data_stream));
+  }
+  uint8_t ReadReg(uint8_t reg) override
+  {
+    uint8_t read_data = 0;
+    i2c.WriteThenRead(kApds9960Address, &reg, 1, &read_data, 1);
+    return read_data;
+  }
+  void Initialize() override
+  {
+    I2cInterface::Status status;
+    i2c.Initialize();
+    status = i2c.Write(kApds9960Address, nullptr, 0);
+    if (status == I2cInterface::Status::kSuccess)
+    {
+      uint8_t initialization_sequence[56] =
+      {
+        kENABLE, kNoMode,
+        kATIME, kAtime,
+        kWTIME, kWtime,
+        kAILTL, kAiltl,
+        kAILTH, kAilth,
+        kAIHTL, kAihtl,
+        kAIHTH, kAihth,
+        kPILT, kPilt,
+        kPIHT, kPiht,
+        kPERS, kPers,
+        kCONFIG1, kConfig1,
+        kPPULSE, kPpulse,
+        kCONTROL, kControl,
+        kCONFIG2, kConfig2,
+        kPoffsetUr, kPoffsetUpRight,
+        kPoffsetDl, kPoffsetDownLeft,
+        kCONFIG3, kConfig3,
+        kGPENTH, kGPEnTh,
+        kGEXTH, kGExTh,
+        kGCONF1, kGconfig1,
+        kGCONF2, kGconfig2,
+        kGoffsetU, kGoffsetUpDown,
+        kGoffsetD, kGoffsetUpDown,
+        kGoffsetL, kGoffsetLeftRight,
+        kGoffsetR, kGoffsetLeftRight,
+        kGPULSE, kGPulse,
+        kGCONF3, kConfig3,
+        kGCONF4, kGconfig4
+      };
+      DEBUG_PRINT("    Found device at address: 0x%02X", kApds9960Address);
+      i2c.Write(kApds9960Address, initialization_sequence,
+        sizeof(initialization_sequence));
+    }
+    else
+    {
+      DEBUG_PRINT("Error Initializing Gesture Sensor");
+    }
+  }
+  bool SetMode(uint8_t mode, bool enable) override
+  {
+    uint8_t reg_value = 0;
+    reg_value = ReadReg(kENABLE);
+
+    if ((mode > 7)) return false;
+    if (enable) {
+      reg_value |= (1 << mode);
+    } else {
+      reg_value &= ~(1 << mode);
+    }
+    WriteReg(kENABLE, reg_value);
+    return true;
+  }
+  void EnableGesture() override
+  {
+    SetMode(0, 0);  // POWEROFF
+
+    WriteReg(kWTIME, 0xFF);
+    WriteReg(kPPULSE, 0x89);
+
+    WriteReg(kCONFIG2, 0xD0);
+
+    SetMode(0, 1);  // POWERON
+    SetMode(3, 1);  // WAIT
+    SetMode(2, 1);  // PROXIMITY
+    SetMode(6, 1);  // GESTURE
+    // kENABLE = 0100'1101
+  }
+  bool DisableGesture() override
+  {
+    if (!SetMode(6, 0)) return false;
+    return true;
+  }
+  uint8_t ReadGestureMode() override
+  {
+    uint8_t value = 0;
+    value = ReadReg(kGCONF4);
+    value &= ~(0xFE);
+    return value;
+  }
+  bool CheckIfGestureOccured() override
+  {
+    uint8_t gstatus = 0;
+    gstatus = ReadReg(kGSTATUS);
+
+    gstatus &= 0b00000001;
+
+    if (gstatus == 1)
```

Comment:
Rather than doing masking, then doing an if/else statement with seperate returns, you could simply do the following:

    return (gstatus & 0b1);

Its a lot cleaner and to the point.